YOLO11 models do not operate with the CUDA and CUDA FP16 targets (OpenCV 4.10.0)

### System Information

OpenCV 4.10.0
Windows 11 Pro
JDK 23.0.1
CUDA 12.4
CuDNN cudnn-windows-x86_64-9.0.0.312_cuda12


### Detailed description

YOLO11 models do not operate with the CUDA and CUDA FP16 targets. Meanwhile, it operates well with the CPU target.


### Steps to reproduce

For instance, the yolo11n.onnx model (https://dropmefiles.com/w332f) throws the following error:

Caused by: CvException [org.opencv.core.CvException: cv::Exception: OpenCV(4.10.0-dev) G:\opencv-4.x\modules\dnn\src\net_impl_fuse.cpp:608: error: (-215:Assertion failed) biasLayerData->outputBlobsWrappers.size() == 1 in function 'cv::dnn::dnn4_v20240521::Net::Impl::fuseLayers'
]
	at org.opencv.dnn.Net.forward_4(Native Method)
	at org.opencv.dnn.Net.forward(Net.java:352)

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [ ] There is reproducer code and related data files (videos, images, onnx, etc)

I have the same problem and I can't find information, I used YOLO11 to do the training and when applying it to opencv in Rust it generates this error.

I even made support attempts directly through the rust opencv lib but they did not return.

Error:

thread 'main' panicked at src\testes\detect_image.rs:72:43:
called `Result::unwrap()` on an `Err` value: Error { code: "StsAssert, -215", message: "OpenCV(4.10.0) E:\\vcpkg\\buildtrees\\opencv4\\src\\4.10.0-e5bca385b2.clean\\modules\\dnn\\src\\net_impl_fuse.cpp:608: error: (-215:Assertion failed) biasLayerData->outputBlobsWrappers.size() == 1 in function 'cv::dnn::dnn4_v20240521::Net::Impl::fuseLayers'\n" }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

```rs
pub fn detect_tree_v2(mat: &mut Mat) -> opencv::Result<()> {

    let mut net = get_mutable_model();

    let input_blob = dnn::blob_from_image(
        mat,
        1.0 / 255.0,
        Size::new(640, 640),    
        opencv::core::Scalar::default(),
        false,
        false,                  
        opencv::core::CV_32F
    )?;

    net.set_input(&input_blob, "images", 1.0, Scalar::default())?;

    let out_layers = net.get_unconnected_out_layers_names()?;
    let mut output = Mat::default();
    
    net.forward(&mut output, &out_layers).unwrap();

    Ok(())
}
```

Issue opencv_rust: https://github.com/twistedfall/opencv-rust/issues/647

Can confirm on 250b5003ee740a6ea4d7523e230128225291fbd2 . Conversion steps:
```bash
wget https://github.com/ultralytics/assets/releases/download/v8.3.0/yolo11n.pt
python3 -m pip install ultralytics
python3 -c "from ultralytics import YOLO; model = YOLO('yolo11n.pt'); model.export(format='onnx')"
```
```
Traceback (most recent call last):
  File "/opencv/run.py", line 10, in <module>
    out = net.forward()
cv2.error: OpenCV(4.12.0-dev) /opencv/modules/dnn/src/net_impl_fuse.cpp:608: error: (-215:Assertion failed) biasLayerData->outputBlobsWrappers.size() == 1 in function 'fuseLayers'
```

--- a/modules/dnn/src/net_impl_fuse.cpp
+++ b/modules/dnn/src/net_impl_fuse.cpp
@@ -370,6 +370,7 @@ void Net::Impl::fuseLayers(const std::vector<LayerPin>& blobsToKeep_)
                 Ptr<NaryEltwiseLayer> nextNaryEltwiseLayer = nextData->layerInstance.dynamicCast<NaryEltwiseLayer>();
                 if (nextEltwiseLayer.empty() && nextNaryEltwiseLayer.empty())
                     break;
+                LayerData *naryOrEltwiseData = nextData;
 
                 // TODO: fused the Conv+NaryEltwise on OpenCL backend. At present, we can only support it at CUDA backend.
                 if (IS_DNN_OPENCL_TARGET(preferableTarget) && nextNaryEltwiseLayer)
@@ -605,8 +606,17 @@ void Net::Impl::fuseLayers(const std::vector<LayerPin>& blobsToKeep_)
                             else if (fuse_eltwise) // conv + eltwise/naryEltwise (note: conv could have fused activations before eltwise)
                             {
                                 CV_Assert(IS_DNN_CUDA_TARGET(preferableTarget));
-                                CV_Assert_N(biasLayerData->outputBlobsWrappers.size() == 1, ld.inputBlobsWrappers.size() == 1);
-                                ld.inputBlobsWrappers.push_back(biasLayerData->outputBlobsWrappers[0]);
+                                CV_Assert_N(biasLayerData->outputBlobsWrappers.size() >= 1, ld.inputBlobsWrappers.size() == 1);
+                                // Iterate over eltwise inputs to find exact output id
+                                for (const auto& pin : naryOrEltwiseData->inputBlobsId)
+                                {
+                                    if (pin.lid == biasLayerData->id)
+                                    {
+                                        ld.inputBlobsWrappers.push_back(biasLayerData->outputBlobsWrappers[pin.oid]);
+                                        break;
+                                    }
+                                }
+                                CV_Assert(ld.inputBlobsWrappers.size() == 2);  // Check input was found
 
                                 if (nextEltwiseLayer)
                                     printf_(("\tfused with %s\n", nextEltwiseLayer->name.c_str()));

